inc/networkportinstantiation.class.php

```diff
@@ -100,11 +100,51 @@ function prepareInputForAdd($input) {
       return parent::prepareInputForAdd($this->prepareInput($input));
    }
 
-
    function prepareInputForUpdate($input) {
       return parent::prepareInputForUpdate($this->prepareInput($input));
    }
 
+   function post_addItem() {
+      $this->manageSocket();
+   }
+
+   function post_updateItem($history = 1) {
+      $this->manageSocket();
+   }
+
+   function manageSocket() {
```

Comment:
I have a lot of trouble understanding how the network part of GLPI works, but this seems very weird to me.
If you change the used Socket, you have to erase previous Socket data and override new Socket data (meaning that you may "steal" it from another item).

Unless I am wrong, Netpoints were made to be used by many items (that is how dropdowns are meant to be used), and now Sockets can be associated to only one item and `NetworkPort`. It looks that it have to be converted into a `CommonDBRelation`, and maybe we should create one Socket for every NetworkPort that is linked to a Netpoint.

Reply:
Unless I am wrong, should use ```CommonDBChild``` like ```Item_Disk``` ?

and maybe rename ```glpi_sokets``` to ```glpi_item_sokets```

Reply:
@trasher any idea ?

Reply:
seen internally (with @orthagh) after meeting, we will use ```CommonDBChild```

Reply:
done